webapp/src/main/java/life/qbic/views/register/RegisterHandler.java

```diff
@@ -0,0 +1,64 @@
+package life.qbic.views.register;
+
+import com.vaadin.flow.component.UI;
+import com.vaadin.flow.component.notification.Notification;
+import life.qbic.usermanagement.User;
+import life.qbic.usermanagement.persistence.UserJpaRepository;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+/**
+ * Handles the {@link RegisterLayout} components
+ *
+ * <p>This class is responsible for enabling buttons or triggering other view relevant changes on
+ * the view class components
+ */
+@Component
+public class RegisterHandler implements RegisterHandlerInterface {
+  private static final org.apache.logging.log4j.Logger log =
+      org.apache.logging.log4j.LogManager.getLogger(RegisterHandler.class);
+
+  private final UserJpaRepository userJpaRepository;
+  private RegisterLayout registeredRegisterLayout;
+
+  RegisterHandler(UserJpaRepository userJpaRepository) {
+    this.userJpaRepository = userJpaRepository;
+  }
+
+  @Override
+  public boolean register(RegisterLayout registerLayout) {
+    if (registeredRegisterLayout != registerLayout) {
+      this.registeredRegisterLayout = registerLayout;
+      // orchestrate view
+      addListener();
+      // then return
+      return true;
+    }
+    return false;
+  }
+
+  private void addListener() {
+    registeredRegisterLayout.registerButton.addClickListener(
+        event -> {
+          List<User> user =
+              userJpaRepository.findUsersByEmail(registeredRegisterLayout.email.getValue());
+
+          if (user.size() > 0) {
```

Comment:
I think here we should go into the model, and not execute this kind of logic in the UI. 

Instead, i suggest to call the use case, make the check there and execute call back methodes via the presenter.